modules/nextflow/src/main/groovy/nextflow/processor/Architecture.groovy

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2023, Pawsey Supercomputing Research Centre
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.processor
+
+import groovy.transform.CompileStatic
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.ToString
+
+/**
+ * Implements the {@code arch} process directive, to hold information on the 
+ * CPU (micro)architecture required by the process.
+ * 
+ * @author Marco De La Pierre <marco.delapierre@gmail.com>
+ */
+@ToString
+@EqualsAndHashCode
+@CompileStatic
+class Architecture {
+
+/*
+ * example of notation in process: arch 'linux/x86_64', target: 'haswell'
+ * example of notation in config:  arch = [name: 'linux/x86_64', target: 'haswell']
+ * 
+ * where dockerArch = 'linux/x86_64'
+ *       spackArch = target ?: arch  // plus some validation for Spack syntax
+ * 
+ *       platform = 'linux'
+ *       arch = 'x86_64'
+ *       target = 'haswell'
+ * 
+ * [alternate example: 'arch linux/arm/v8', where platform = 'linux' and arch = 'arm/v8']
+ */
+    // used in Nextflow
+    final String dockerArch
+    final String spackArch
+
+    // defined, but currently not used
+    final String platform
+    final String arch
+    final String target
+
+    @CompileStatic
```

Comment:
Since the class is marked `@CompileStatic`, this is not needed here

Reply:
done!